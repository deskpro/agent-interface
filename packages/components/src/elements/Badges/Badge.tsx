import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/elements/tag-badge.css";

export type BadgeProps = {
  color?: "danger" | "warning" | "warningLight" | "empty" | "yellow" | "white";
};

const Badge: React.SFC<BadgeProps> = ({ children, color }) => (
  <span
    className={classNames("dp-Badge", {
      [`Badge--${color}`]: !!color
    })}
  >
    {children}
  </span>
);

export default Badge;
