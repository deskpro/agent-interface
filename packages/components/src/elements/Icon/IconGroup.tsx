import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/components/dp-ListPane.css";

export type IconGroupProps = {
  className?: string;
};

const IconGroup: React.SFC<IconGroupProps> = ({ children, className }) => (
  <span className={classNames("dp-IconGroup", className)}>{children}</span>
);

export default IconGroup;
