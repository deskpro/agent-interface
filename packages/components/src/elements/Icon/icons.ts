import { ReactComponent as Acrobat } from "../../../icons/Acrobat.svg";
import { ReactComponent as AddFollower } from "../../../icons/AddFollower.svg";
import { ReactComponent as AddIM } from "../../../icons/AddIM.svg";
import { ReactComponent as AddTab } from "../../../icons/AddTab.svg";
import { ReactComponent as AddUser } from "../../../icons/AddUser.svg";
import { ReactComponent as Admin } from "../../../icons/Admin.svg";
import { ReactComponent as Approve } from "../../../icons/Approve.svg";
import { ReactComponent as Assets } from "../../../icons/Assets.svg";
import { ReactComponent as Attach } from "../../../icons/Attach.svg";
import { ReactComponent as Avatar } from "../../../icons/Avatar.svg";
import { ReactComponent as Ban } from "../../../icons/Ban.svg";
import { ReactComponent as Beetle } from "../../../icons/Beetle.svg";
import { ReactComponent as Billing } from "../../../icons/Billing.svg";
import { ReactComponent as BillingSidebar } from "../../../icons/BillingSidebar.svg";
import { ReactComponent as Bin } from "../../../icons/Bin.svg";
import { ReactComponent as Bolt } from "../../../icons/Bolt.svg";
import { ReactComponent as Calendar } from "../../../icons/Calendar.svg";
import { ReactComponent as CalendarSimple } from "../../../icons/CalendarSimple.svg";
import { ReactComponent as Caret } from "../../../icons/Caret.svg";
import { ReactComponent as CaretDown } from "../../../icons/CaretDown.svg";
import { ReactComponent as CaretLeft } from "../../../icons/CaretLeft.svg";
import { ReactComponent as CaretUp } from "../../../icons/CaretUp.svg";
import { ReactComponent as CaretRight } from "../../../icons/CaretRight.svg";
import { ReactComponent as CaretFilledDown } from "../../../icons/CaretFilledDown.svg";
import { ReactComponent as CaretFilledUp } from "../../../icons/CaretFilledUp.svg";
import { ReactComponent as Chat } from "../../../icons/Chat.svg";
import { ReactComponent as Close } from "../../../icons/Close.svg";
import { ReactComponent as Cloud } from "../../../icons/Cloud.svg";
import { ReactComponent as Copy } from "../../../icons/Copy.svg";
import { ReactComponent as CreateTicket } from "../../../icons/CreateTicket.svg";
import { ReactComponent as Criteria } from "../../../icons/Criteria.svg";
import { ReactComponent as CRM } from "../../../icons/CRM.svg";
import { ReactComponent as CSV } from "../../../icons/CSV.svg";
import { ReactComponent as Cursor } from "../../../icons/Cursor.svg";
import { ReactComponent as Deal } from "../../../icons/Deal.svg";
import { ReactComponent as Delete } from "../../../icons/Delete.svg";
import { ReactComponent as Document } from "../../../icons/Document.svg";
import { ReactComponent as Download } from "../../../icons/Download.svg";
import { ReactComponent as Drag } from "../../../icons/Drag.svg";
import { ReactComponent as Edit } from "../../../icons/Edit.svg";
import { ReactComponent as Email } from "../../../icons/Email.svg";
import { ReactComponent as EndCall } from "../../../icons/EndCall.svg";
import { ReactComponent as FacetedBy } from "../../../icons/FacetedBy.svg";
import { ReactComponent as Feedback } from "../../../icons/Feedback.svg";
import { ReactComponent as Find } from "../../../icons/Find.svg";
import { ReactComponent as Following } from "../../../icons/Following.svg";
import { ReactComponent as FollowUp } from "../../../icons/FollowUp.svg";
import { ReactComponent as Forms } from "../../../icons/Forms.svg";
import { ReactComponent as CloseFill } from "../../../icons/CloseFill.svg";
import { ReactComponent as Global } from "../../../icons/Global.svg";
import { ReactComponent as GroupedBy } from "../../../icons/GroupedBy.svg";
import { ReactComponent as History } from "../../../icons/History.svg";
import { ReactComponent as Hold } from "../../../icons/Hold.svg";
import { ReactComponent as Info } from "../../../icons/Info.svg";
import { ReactComponent as Insert } from "../../../icons/Insert.svg";
import { ReactComponent as InsertLink } from "../../../icons/InsertLink.svg";
import { ReactComponent as Jump } from "../../../icons/Jump.svg";
import { ReactComponent as KanbanView } from "../../../icons/KanbanView.svg";
import { ReactComponent as Link } from "../../../icons/Link.svg";
import { ReactComponent as LinkedTickets } from "../../../icons/LinkedTickets.svg";
import { ReactComponent as Liscense } from "../../../icons/Liscense.svg";
import { ReactComponent as ListView } from "../../../icons/ListView.svg";
import { ReactComponent as Lock } from "../../../icons/Lock.svg";
import { ReactComponent as Menu } from "../../../icons/Menu.svg";
import { ReactComponent as Merge } from "../../../icons/Merge.svg";
import { ReactComponent as Mute } from "../../../icons/Mute.svg";
import { ReactComponent as New } from "../../../icons/New.svg";
import { ReactComponent as News } from "../../../icons/News.svg";
import { ReactComponent as Notifications } from "../../../icons/Notifications.svg";
import { ReactComponent as OnHold } from "../../../icons/Onhold.svg";
import { ReactComponent as Organisation } from "../../../icons/Organisation.svg";
import { ReactComponent as Owner } from "../../../icons/Owner.svg";
import { ReactComponent as Pending } from "../../../icons/Pending.svg";
import { ReactComponent as Phone } from "../../../icons/Phone.svg";
import { ReactComponent as PhoneFilled } from "../../../icons/PhoneFilled.svg";
import { ReactComponent as Play } from "../../../icons/Play.svg";
import { ReactComponent as Portal } from "../../../icons/Portal.svg";
import { ReactComponent as Print } from "../../../icons/Print.svg";
import { ReactComponent as Publish } from "../../../icons/Publish.svg";
import { ReactComponent as Question } from "../../../icons/Question.svg";
import { ReactComponent as Record } from "../../../icons/Record.svg";
import { ReactComponent as Reports } from "../../../icons/Reports.svg";
import { ReactComponent as Rocket } from "../../../icons/Rocket.svg";
import { ReactComponent as Scissors } from "../../../icons/Scissors.svg";
import { ReactComponent as Security } from "../../../icons/Security.svg";
import { ReactComponent as Settings } from "../../../icons/Settings.svg";
import { ReactComponent as Snippet } from "../../../icons/Snippet.svg";
import { ReactComponent as Sort } from "../../../icons/Sort.svg";
import { ReactComponent as Star } from "../../../icons/Star.svg";
import { ReactComponent as SubField } from "../../../icons/SubField.svg";
import { ReactComponent as Support } from "../../../icons/Support.svg";
import { ReactComponent as Swap } from "../../../icons/Swap.svg";
import { ReactComponent as TableView } from "../../../icons/TableView.svg";
import { ReactComponent as Tasks } from "../../../icons/Tasks.svg";
import { ReactComponent as TasksSidebar } from "../../../icons/TasksSidebar.svg";
import { ReactComponent as Tick } from "../../../icons/Tick.svg";
import { ReactComponent as TickDottedLine } from "../../../icons/TickDottedLine.svg";
import { ReactComponent as Ticket } from "../../../icons/Ticket.svg";
import { ReactComponent as TicketProperties } from "../../../icons/TicketProperties.svg";
import { ReactComponent as TicketTime } from "../../../icons/TicketTime.svg";
import { ReactComponent as Transfer } from "../../../icons/Transfer.svg";
import { ReactComponent as TypeSize } from "../../../icons/TypeSize.svg";
import { ReactComponent as Unlock } from "../../../icons/Unlock.svg";
import { ReactComponent as User } from "../../../icons/User.svg";
import { ReactComponent as UserAwaiting } from "../../../icons/UserAwaiting.svg";
import { ReactComponent as View } from "../../../icons/View.svg";
import { ReactComponent as Visibility } from "../../../icons/Visibility.svg";
import { ReactComponent as Voice } from "../../../icons/Voice.svg";
import { ReactComponent as VoiceCall } from "../../../icons/VoiceCall.svg";
import { ReactComponent as Warning } from "../../../icons/Warning.svg";

export default {
  acrobat: Acrobat,
  "add-follower": AddFollower,
  "add-im": AddIM,
  "add-tab": AddTab,
  "add-user": AddUser,
  admin: Admin,
  approve: Approve,
  assets: Assets,
  attach: Attach,
  avatar: Avatar,
  ban: Ban,
  beetle: Beetle,
  billing: Billing,
  "billing-sidebar": BillingSidebar,
  bin: Bin,
  bolt: Bolt,
  calendar: Calendar,
  "calendar-simple": CalendarSimple,
  caret: Caret,
  "caret-down": CaretDown,
  "caret-left": CaretLeft,
  "caret-up": CaretUp,
  "caret-right": CaretRight,
  "caret-filled-down": CaretFilledDown,
  "caret-filled-up": CaretFilledUp,
  chat: Chat,
  close: Close,
  cloud: Cloud,
  copy: Copy,
  "create-ticket": CreateTicket,
  criteria: Criteria,
  crm: CRM,
  csv: CSV,
  cursor: Cursor,
  deal: Deal,
  delete: Delete,
  document: Document,
  download: Download,
  drag: Drag,
  edit: Edit,
  email: Email,
  "end-call": EndCall,
  "faceted-by": FacetedBy,
  feedback: Feedback,
  find: Find,
  following: Following,
  "follow-up": FollowUp,
  forms: Forms,
  "close-fill": CloseFill,
  global: Global,
  "grouped-by": GroupedBy,
  history: History,
  hold: Hold,
  info: Info,
  insert: Insert,
  "insert-link": InsertLink,
  jump: Jump,
  "kanban-view": KanbanView,
  link: Link,
  "linked-tickets": LinkedTickets,
  liscense: Liscense,
  "list-view": ListView,
  lock: Lock,
  menu: Menu,
  merge: Merge,
  mute: Mute,
  new: New,
  news: News,
  notifications: Notifications,
  "on-hold": OnHold,
  organisation: Organisation,
  owner: Owner,
  pending: Pending,
  phone: Phone,
  "phone-filled": PhoneFilled,
  play: Play,
  portal: Portal,
  print: Print,
  publish: Publish,
  question: Question,
  record: Record,
  reports: Reports,
  rocket: Rocket,
  scissors: Scissors,
  security: Security,
  settings: Settings,
  snippet: Snippet,
  sort: Sort,
  star: Star,
  subfield: SubField,
  support: Support,
  swap: Swap,
  "table-view": TableView,
  tasks: Tasks,
  "tasks-sidebar": TasksSidebar,
  tick: Tick,
  "tick-dotted-line": TickDottedLine,
  ticket: Ticket,
  "ticket-properties": TicketProperties,
  "ticket-time": TicketTime,
  transfer: Transfer,
  "type-size": TypeSize,
  unlock: Unlock,
  user: User,
  "user-awaiting": UserAwaiting,
  view: View,
  visibility: Visibility,
  voice: Voice,
  "voice-call": VoiceCall,
  warning: Warning
};
