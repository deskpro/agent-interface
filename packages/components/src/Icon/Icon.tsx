import * as React from "react";
import "@deskpro/agent-interface-style/dist/elements/dp-Icons.css";
import icons from "./icons";

export type IconProps = {
  name: string;
  size?: number;
  color?: "default" | "primary" | "dark";
};

const colors = {
  default: "#979797",
  primary: "#3a8dde",
  dark: "#8B9293"
};

const Icon: React.SFC<IconProps> = ({ name, size = 24, color = "default" }) => {
  const SvgIcon = icons[name];
  if (!SvgIcon) {
    return null;
  }
  return (
    <span className={`dp-Icon dp-Icon--${name}`}>
      <SvgIcon
        fill={colors[color] || colors.default}
        width={size}
        height={size}
      />
    </span>
  );
};

Icon.defaultProps = {
  color: "default",
  size: 24
};

export default Icon;
