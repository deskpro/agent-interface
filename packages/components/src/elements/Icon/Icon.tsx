import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/elements/dp-Icons.css";
import icons from "./icons";

export type IconProps = React.ComponentProps<"span"> & {
  name: string;
  size?: number;
  circle?: boolean;
  color?: DPColor;
  iconRef?: React.Ref<HTMLSpanElement>;
  onClick?(
    e: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>
  ): void;
};

const Icon: React.FC<IconProps> = ({
  name,
  size = 24,
  color,
  circle,
  iconRef,
  onClick,
  ...props
}) => {
  const SvgIcon = icons[name];
  if (!SvgIcon) {
    return null;
  }
  return (
    <span
      className={classNames(`dp-Icon Icon--${name}`, {
        "Icon--circle": circle,
        [`Icon--${color}`]: !!color
      })}
      role="button"
      tabIndex={-1}
      onClick={onClick}
      onKeyDown={onClick}
      ref={iconRef}
      {...props}
    >
      <SvgIcon width={size} height={size} />
    </span>
  );
};

Icon.defaultProps = {
  color: undefined,
  size: 24
};

export default React.memo(
  React.forwardRef<HTMLSpanElement, IconProps>((props, ref) => (
    <Icon {...props} iconRef={ref as React.Ref<HTMLSpanElement>} />
  ))
);
