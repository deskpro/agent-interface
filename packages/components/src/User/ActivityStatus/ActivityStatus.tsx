import * as React from "react";
import classNames from "classnames";

import Icon from "../../elements/Icon/Icon";

import "@deskpro/agent-interface-style/dist/components/dp-icons.css";

export type ActivityStatusProps = {
  status: "writing" | "viewing";
  color: DPColor;
  className?: string;
};

const ActivityStatus: React.SFC<ActivityStatusProps> = ({
  status,
  color,
  children,
  className
}) => (
  <span
    className={classNames(
      "dp-ActivityStatus",
      `ActivityStatus--${status}`,
      className,
      { [`ActivityStatus--${color}`]: !!color }
    )}
  >
    <Icon
      name={status === "writing" ? "edit" : "view"}
      color={color}
      size={15}
    />
    {children}
  </span>
);

export default ActivityStatus;
