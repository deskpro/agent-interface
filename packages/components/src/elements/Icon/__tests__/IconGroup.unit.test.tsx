import * as React from "react";
import { render } from "react-testing-library";

import IconGroup from "../IconGroup";
import Icon from "../Icon";
import Badge from "../../Badges/Badge";
import ActivityStatus from "../../../User/ActivityStatus/ActivityStatus";

describe("<IconGroup />", () => {
  it("should match is snapshot", () => {
    const { container } = render(
      <IconGroup className="icon-group">
        <ActivityStatus status="viewing">
          <Icon name="avatar" size={16} />
          <Icon name="avatar" size={16} />
        </ActivityStatus>
        <Icon name="beetle" size={18} circle />
        <Badge color="grey-light" type="round" />
        <Badge color="purple" type="round" />
      </IconGroup>
    );
    expect(container.querySelector(".icon-group")).toMatchSnapshot();
  });
});
