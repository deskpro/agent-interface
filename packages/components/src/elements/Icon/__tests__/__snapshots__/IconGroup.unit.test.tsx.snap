// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconGroup /> should match is snapshot 1`] = `
<span
  class="dp-IconGroup icon-group"
>
  <span
    class="dp-ActivityStatus ActivityStatus--viewing"
  />
  <span
    class="dp-Badge Badge--round Badge--grey-light"
  >
    
  </span>
  <span
    class="dp-Badge Badge--round Badge--purple"
  >
    
  </span>
</span>
`;
